File: unit2/hello_name_age.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    if (argc != 3)
    {
        printf("Usage: ./hello_name_age <first name> <age>\n");
        return 1;
    }

    int age = atoi(argv[2]);

    if (age < 0)
    {
        printf("Usage: age must be a non-negative integer.\n");
        return 2;
    }

    printf("hello, %s!\n", argv[1]);
    printf("On your next birthday, you will be %d.\n", age + 1);

    return 0;
}

File: unit3/hello_name_age.py
from sys import argv, exit

def main():
    argc = len(argv)

    if argc != 3:
        print("Usage: ./hello_name_age <first name> <age>")
        exit(1)

    age = int(argv[2])

    if age < 0:
        print("Usage: age must be a non-negative integer.")
        exit(2)

    print(f"hello, {argv[1]}!")
    print(f"On your next birthday, you will be {age + 1}.")

if __name__ == "__main__":
    main()